BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
+             {
+                 if ( result.isCancelled() )
+                 {
+-                    disconnect( result.getCancelReasonComponents() );
++                    // Waterfall start - we're going on an adventure!
++//                    disconnect( result.getCancelReasonComponents() );
++                    disconnect(result.cancelReason());
++                    // Waterfall end
+                     return;
+                 }
+                 if ( ch.isClosed() )
+@@ -529,7 +542,10 @@ public class InitialHandler extends PacketHandler implements PendingConnection
+             {
+                 if ( result.isCancelled() )
+                 {
+-                    disconnect( result.getCancelReasonComponents() );
++                    // Waterfall start - we're going on an adventure!
++//                    disconnect( result.getCancelReasonComponents() );
++                    disconnect(result.cancelReason());
++                    // Waterfall end
+                     return;
+                 }
+                 if ( ch.isClosed() )
+@@ -590,7 +606,10 @@ public class InitialHandler extends PacketHandler implements PendingConnection
+     {
+         if ( canSendKickMessage() )
+         {
+-            disconnect( TextComponent.fromLegacyText( reason ) );
++            // Waterfall start - we're going on an adventure!
++//            disconnect( TextComponent.fromLegacyText( reason ) );
++            disconnect(LegacyComponentSerializer.legacySection().deserialize(reason));
++            // Waterfall end
+         } else
+         {
+             ch.close();
+@@ -600,13 +619,16 @@ public class InitialHandler extends PacketHandler implements PendingConnection
+     @Override
+     public void disconnect(final BaseComponent... reason)
+     {
+-        if ( canSendKickMessage() )
+-        {
+-            ch.delayedClose( new Kick( ComponentSerializer.toString( reason ) ) );
+-        } else
+-        {
+-            ch.close();
+-        }
++        // Waterfall start - we're going on an adventure!
++//        if ( canSendKickMessage() )
++//        {
++//            ch.delayedClose( new Kick( ComponentSerializer.toString( reason ) ) );
++//        } else
++//        {
++//            ch.close();
++//        }
++        disconnect(BungeeComponentSerializer.get().deserialize(reason));
++        // Waterfall end
+     }
+ 
+     @Override
+@@ -618,7 +640,17 @@ public class InitialHandler extends PacketHandler implements PendingConnection
+         } );
+     }
+ 
++    // Waterfall start - we're going on an adventure!
+     @Override
++    public void disconnect(Component reason) {
++        if (canSendKickMessage()) {
++            ch.delayedClose(new Kick(GsonComponentSerializer.gson().serialize(reason)));
++        } else {
++            ch.close();
++        }
++    }
++    // Waterfall end
++
+     public String getName()
+     {
```

Comment:
You stole the `@Override` from `getName()` :o

Reply:
Yeah IntelliJ tricked me there haha. It told me that the `@Override` on `getName` was newly added, and so I thought I might've accidentally added that lol